.github/workflows/publish-docs.yaml

```diff
@@ -0,0 +1,53 @@
+name: Publish Documentation
+
+on:
+  push:
+    branches:
+      - documentation/S3CSI-17-setup-github-pages
+  workflow_dispatch:
+
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+concurrency:
+  group: "pages"
+  cancel-in-progress: false
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    environment:
+      name: github-pages
+      url: ${{ steps.deployment.outputs.page_url }}
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          lfs: true
+
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.x
+          cache: pip
+
+      - name: Install dependencies
+        run: pip install -r requirements.txt
+
+      - name: Build MkDocs site
+        run: mkdocs build --strict
```

Comment:
Do it in another PR, but add a `mkdocs build --strict` in the pre merge phase to make sure you have no surprise when the times come to deploy it.